### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by PabloGrok** [darkbrain-fc/HardwareSupervisor#8]
Is there any documentation to see how this works?

I got the service installed and running, but I have no idea where it's storing the log files, where I should put the config.yaml file, what's the syntax of that config.yaml file, etc. I can't even see where the executable is (if there's one...)

The `get-wmiobject -namespace "root/HardwareSupervisor" -Class Sensor` line doesn't work yielding this error:

```
get-wmiobject: The term 'get-wmiobject' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.

```

I haven't tried rainmeter, but that's because I don't want to use RM. My final goal is to feed data from this service into Telegraf and go from there.

Can I have a web server providing data like in LibreOfficeMonitor?

Thanks for your help.

EDIT: It seems Get-WmiObject was deprecated long time ago and is not present in PowerShell Core, but still works in the original PowerShell that's installed on Windows (the HardwareSupervisor service is working in my system and I got data from running the line on the old Powershell). I got it to work, but shouldn't there be a way to use that query with up to date commands?

**Comment by darkbrain-fc** [comment]
> Is there any documentation to see how this works?
You're right there is no much documentation for now. Anyway HardwareSupervisor is a simple Service built around LibreHardwareMonitor. BTW I'll add some information in the near future.
 
> I got the service installed and running, but I have no idea where it's storing the log files, where I should put the config.yaml file, what's the syntax of that config.yaml file, etc. I can't even see where the executable is (if there's one...)
You should find everything here: `C:\Program Files (x86)\HardwareSupervisor`

> I haven't tried rainmeter, but that's because I don't want to use RM. My final goal is to feed data from this service into Telegraf and go from there.
Hmm quite interesting! If you need some help to write a script to query HardwareSupervisor write me. We could write a simple C# handler to send data to Telegraf.
 
> Can I have a web server providing data like in LibreOfficeMonitor?
I can but i don't want to. WMI is a protocol build over HTTP so it's should be enough (https://docs.microsoft.com/en-us/windows/win32/wmisdk/about-wmi) even if I didn't try it should be queried from another host in WMI without too many issues.
 
> EDIT: It seems Get-WmiObject was deprecated long time ago and is not present in PowerShell Core, but still works in the original PowerShell that's installed on Windows (the HardwareSupervisor service is working in my system and I got data from running the line on the old Powershell). I got it to work, but shouldn't there be a way to use that query with up to date commands?
OMG I didn't know that! I will looking for an alternative.
 
> EDIT2: Although I find Rainmeter a mess I' not willing to learn and I didn't get the plugin to work, the Configurator.exe helper is a good way to check if HardwareSupervisor is running.
Yes Rainmeter is cool but too far complex.

**Comment by PabloGrok** [comment]
Thank you for your thorough reply.

For some reason I didn't think this was a 32 bit app, so I didn't search in Program Files (x86). Duh!

Regarding Telegraf, I think most of the Network Monitoring Systems understand many formats and I bet WMI is supported in all of them, so you're halfway there for the most popular ones (I'm thinking Prometheus, Zabbix, Graphite or Influxdb etc.). I'll be happy to help if you want to implement something like this. I'm currently in the phase of trying a bunch of these NMSs, so I'll hopefully soon be able to understand what they require.

Looking forward to some documentation (especially on the config.yaml file syntax)!
